### PR TITLE
Remove the deprecated `LightningDataModule.test_transforms` 

### PR DESCRIPTION
**Pull request by rschireman**
## What does this PR do?

Fixes part of #12521 by removing the deprecated LightningDataModule.test_transforms 

### Does your PR introduce any breaking changes? If yes, please list them.

Removes deprecated `LightningDataModule.test_transforms`

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -118,6 +118,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed support for passing strategy names or strategy instances to the plugins Trainer argument ([#12700](https://github.com/PyTorchLightning/pytorch-lightning/pull/12700))
 
 
+- Removed the deprecated `test_transforms` argument from the `LightningDataModule` constructor ([#12763](https://github.com/PyTorchLightning/pytorch-lightning/pull/12773))
```
Comment (akihironitta) [inline CHANGELOG.md:121]:
```suggestion
- Removed the deprecated `test_transforms` argument from the `LightningDataModule` constructor ([#12773](https://github.com/PyTorchLightning/pytorch-lightning/pull/12773))
```

File: docs/source/starter/core_guide.rst [hunk 2]
```diff
@@ -1139,7 +1139,6 @@ spread all over files.
     # transforms
     train_transforms = ...
     val_transforms = ...
-    test_transforms = ...
```
Comment (akihironitta) [inline docs/source/starter/core_guide.rst:1142]: This line doesn't need to be deleted, I think?

Comment (rohitgr7) [reply docs/source/starter/core_guide.rst:1142]: no.

File: docs/source/starter/core_guide.rst [hunk 3]
```diff
@@ -1139,7 +1139,6 @@ spread all over files.
     # transforms
     train_transforms = ...
     val_transforms = ...
```
Comment (rohitgr7) [inline docs/source/starter/core_guide.rst:1141]:
```suggestion
    val_transforms = ...
    test_transforms = ...
```

**Review by akihironitta — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: docs/source/starter/core_guide.rst (hunk 3 above)
Comment (Xl5843003) [reply docs/source/starter/core_guide.rst]: -    val_transforms = ...
+    val_transforms = ...
+    test_transforms = ...

**Pull request merged**